.github/workflows/ci-master.yml

```diff
@@ -3,63 +3,184 @@ on:
+        mv $SOURCE_ARTIFACT/src/{zcoin-cli,zcoin-tx,zcoind,qt/zcoin-qt} $ARTIFACT_DIR
+    - name: Upload Artifact
+      uses: actions/upload-artifact@v1
+      with:
+        name: linux-binaries
+        path: ${{ env.ARTIFACT_DIR }}
+    - name: Prepare Test Logs for Artifact
+      if: failure()
+      run: |
+        tor_log=$SOURCE_ARTIFACT/src/tor/test-suite.log
+        zcoin_log=$SOURCE_ARTIFACT/src/test-suite.log
 
-  target-windows:
-    name: Target Windows
-    runs-on: ubuntu-latest
+        mkdir -p $TEST_LOG_ARTIFACT_DIR
+
+        if test -f $tor_log; then
+          mv $tor_log $TEST_LOG_ARTIFACT_DIR/tor.log
+        fi
+
+        if test -f $zcoin_log; then
+          mv $zcoin_log $TEST_LOG_ARTIFACT_DIR/zcoin.log
+        fi
+    - name: Upload Test Logs Artifact
+      if: failure()
+      uses: actions/upload-artifact@v1
+      with:
+        name: test-logs
+        path: ${{ env.TEST_LOG_ARTIFACT_DIR }}
+  build-windows:
+    name: Build for Windows
+    needs: create-source-distribution
+    runs-on: ubuntu-18.04
+    env:
+      ARTIFACT_DIR: windows-binaries
     steps:
-    - name: Checkout
-      uses: actions/checkout@v1
-    - name: Install dependencies
+    - name: Getting Source
+      uses: actions/download-artifact@v1
+      with:
+        name: ${{ env.SOURCE_ARTIFACT }}
+    - name: Extract Archives
       run: |
-        sudo apt update
-        sudo apt install -y ccache g++-mingw-w64-x86-64 mingw-w64-x86-64-dev
+        tar -xzf depends.tar.gz
+        tar -xzf zcoin-*.tar.gz --strip-components=1
+      working-directory: ${{ env.SOURCE_ARTIFACT }}
+    - name: Install Required Packages
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y g++-mingw-w64-x86-64 gcc-mingw-w64-x86-64
     - name: Switch MinGW GCC and G++ to POSIX Threading
       run: |
         sudo update-alternatives --set x86_64-w64-mingw32-gcc /usr/bin/x86_64-w64-mingw32-gcc-posix
         sudo update-alternatives --set x86_64-w64-mingw32-g++ /usr/bin/x86_64-w64-mingw32-g++-posix
-    - name: Build dependencies
+    - name: Build Dependencies
       run: make -C depends -j$(nproc) HOST=x86_64-w64-mingw32
+      working-directory: ${{ env.SOURCE_ARTIFACT }}
     - name: Build Zcoin
       run: |
-        ./autogen.sh
-        ./configure --disable-jni --enable-exodus --prefix=$(pwd)/depends/x86_64-w64-mingw32
+        ./configure --disable-jni --enable-exodus --prefix=$(realpath depends/x86_64-w64-mingw32)
         make -j$(nproc)
-
-  target-macos:
-    name: Target macOS
-    runs-on: macos-latest
+      working-directory: ${{ env.SOURCE_ARTIFACT }}
+    - name: Prepare Files for Artifact
+      run: |
+        mkdir -p $ARTIFACT_DIR
+        mv $SOURCE_ARTIFACT/src/{zcoin-cli.exe,zcoin-tx.exe,zcoind.exe,qt/zcoin-qt.exe} $ARTIFACT_DIR
+    - name: Upload Artifact
+      uses: actions/upload-artifact@v1
+      with:
+        name: windows-binaries
+        path: ${{ env.ARTIFACT_DIR }}
+  build-mac:
```

Comment:
It should be macos if we refer to the OS.

Reply:
I think just `mac` is enough. It not ambiguous and shorter to type.

Reply:
Hmm, Ok makes sense.